runtime/Java/src/org/antlr/v4/runtime/UnbufferedCharStream.java

```diff
@@ -82,7 +82,7 @@ public UnbufferedCharStream() {
 	/** Useful for subclasses that pull char from other than this.input. */
 	public UnbufferedCharStream(int bufferSize) {
 		n = 0;
-		data = new char[bufferSize];
+		data = new int[bufferSize];
 	}
 
 	public UnbufferedCharStream(InputStream input) {
```

Comment:
Sure, I'll make the changes! Nope, we'll still use a `Reader`, we'll just use a different `InputStreamReader` constructor which allows us to specify a `Charset`.